src/app/seamlessupdate/client/Settings.java

```diff
@@ -11,23 +10,35 @@
 import android.preference.PreferenceManager;
 
 public class Settings extends PreferenceActivity {
-    private static final int DEFAULT_NETWORK_TYPE = JobInfo.NETWORK_TYPE_ANY;
+    private static final String KEY_CHANNEL = "channel";
     private static final String KEY_NETWORK_TYPE = "network_type";
-    static final String KEY_BATTERY_NOT_LOW = "battery_not_low";
-    static final String KEY_IDLE_REBOOT = "idle_reboot";
+    private static final String KEY_BATTERY_NOT_LOW = "battery_not_low";
+    private static final String KEY_IDLE_REBOOT = "idle_reboot";
     static final String KEY_WAITING_FOR_REBOOT = "waiting_for_reboot";
 
     static SharedPreferences getPreferences(final Context context) {
         final Context deviceContext = context.createDeviceProtectedStorageContext();
         return PreferenceManager.getDefaultSharedPreferences(deviceContext);
     }
 
+    static String getChannel(final Context context) {
+        String def = context.getString(R.string.channel_default);
+        return getPreferences(context).getString(KEY_CHANNEL, def);
+    }
+
```

Comment:
Unfortunately, we need those methods for settings access, because `PreferenceManager#setDefaultValues()` is a UI method that doesn't affect the service which might start before the user opens the settings screen.